cpp/core/jni/JniCommon.h

```diff
@@ -348,47 +343,30 @@ class SparkAllocationListener final : public gluten::AllocationListener {
   }
 
  private:
-  int64_t reserve(int64_t diff) {
-    std::lock_guard<std::mutex> lock(mutex_);
-    bytesReserved_ += diff;
-    int64_t newBlockCount;
-    if (bytesReserved_ == 0) {
-      newBlockCount = 0;
-    } else {
-      // ceil to get the required block number
-      newBlockCount = (bytesReserved_ - 1) / blockSize_ + 1;
-    }
-    int64_t bytesGranted = (newBlockCount - blocksReserved_) * blockSize_;
-    blocksReserved_ = newBlockCount;
-    maxBytesReserved_ = std::max(maxBytesReserved_, bytesReserved_);
-    return bytesGranted;
-  }
```

Comment:
Does `VeloxMemoryManager::arrowPool_` still use this code? Can we move it into another independent `gluten::AllocationListener` implementation which can wrap another listener? Like the way `BacktraceAllocationListener` adopted.

Reply:
I move such logic to ListenableMemoryAllocator.